TestSyncLoadOnObjectWhichCanNotFoundInStorage errors like nil pointer dereference

## Bug Report

Please answer these questions before submitting your issue. Thanks!

### 1. Minimal reproduce step (Required)

<!-- a step by step guide for reproducing the bug. -->
https://do.pingcap.net/jenkins/blue/organizations/jenkins/pingcap%2Ftidb%2Fghpr_unit_test/detail/ghpr_unit_test/24158/tests

### 2. What did you expect to see? (Required)
TestSyncLoadOnObjectWhichCanNotFoundInStorage test sucessfully

### 3. What did you see instead (Required)
```
--- FAIL: TestSyncLoadOnObjectWhichCanNotFoundInStorage (51.04s)
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
	panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x8679bb8]
goroutine 49 [running]:
testing.tRunner.func1.2({0x8d2e500, 0xd0e4620})
	GOROOT/src/testing/testing.go:1632 +0x3fc
testing.tRunner.func1()
	GOROOT/src/testing/testing.go:1635 +0x6b6
panic({0x8d2e500?, 0xd0e4620?})
	GOROOT/src/runtime/panic.go:785 +0x132
pkg/statistics/handle/syncload/syncload_test_test.TestSyncLoadOnObjectWhichCanNotFoundInStorage(0xc00120d520)
	pkg/statistics/handle/syncload/stats_syncload_test.go:469 +0x12d8
testing.tRunner(0xc00120d520, 0x96df748)
	GOROOT/src/testing/testing.go:1690 +0x227
created by testing.(*T).Run in goroutine 1
	GOROOT/src/testing/testing.go:1743 +0x826
```

### 4. What is your TiDB version? (Required)

<!-- Paste the output of SELECT tidb_version() -->
master

--- a/pkg/statistics/handle/syncload/stats_syncload_test.go
+++ b/pkg/statistics/handle/syncload/stats_syncload_test.go
@@ -448,8 +448,11 @@ func TestSyncLoadOnObjectWhichCanNotFoundInStorage(t *testing.T) {
 
 	// Try sync load.
 	tk.MustExec("select * from t where a >= 1 and b = 2 and c = 3 and d = 4")
-	statsTbl, ok = h.Get(tblInfo.ID)
-	require.True(t, ok)
+	require.Eventually(t, func() bool {
+		statsTbl, ok = h.Get(tblInfo.ID)
+		require.True(t, ok)
+		return statsTbl.ColNum() == 3
+	}, 5*time.Second, 100*time.Millisecond)
 	require.True(t, statsTbl.GetCol(tblInfo.Columns[0].ID).IsFullLoad())
 	require.True(t, statsTbl.GetCol(tblInfo.Columns[1].ID).IsFullLoad())
 	require.True(t, statsTbl.GetCol(tblInfo.Columns[3].ID).IsFullLoad())
@@ -463,8 +466,11 @@ func TestSyncLoadOnObjectWhichCanNotFoundInStorage(t *testing.T) {
 	tk.MustExec("analyze table t columns a, b, c")
 	require.NoError(t, h.InitStatsLite(context.TODO()))
 	tk.MustExec("select * from t where a >= 1 and b = 2 and c = 3 and d = 4")
-	statsTbl, ok = h.Get(tblInfo.ID)
-	require.True(t, ok)
+	require.Eventually(t, func() bool {
+		statsTbl, ok = h.Get(tblInfo.ID)
+		require.True(t, ok)
+		return statsTbl.ColNum() == 4
+	}, 5*time.Second, 100*time.Millisecond)
 	// a, b, d's status is not changed.
 	require.True(t, statsTbl.GetCol(tblInfo.Columns[0].ID).IsFullLoad())
 	require.True(t, statsTbl.GetCol(tblInfo.Columns[1].ID).IsFullLoad())

